.github/workflows/copyright-check.yml

```
@@ -0,0 +1,59 @@
+# Copyright (c) 2020-2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: Copyright check
+
+on: 
+  pull_request:
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          path: ${{ github.run_id }}
+          fetch-depth: 0
+
+      - name: Run on change-set
```

This step does not run on the change-set, it actually scans all files for missing copy-right note. But I think that's not an issue, since we already know that all files in `main` have this notice so for new PRs it's fine to check all files. How about we name this step like this instead?:

```suggestion
      - name: All files have copyright notice
```

---

SG good suggestion